src/components/Page/components/Header/Header.tsx

```diff
@@ -81,6 +81,14 @@ export function Header({
 }: HeaderProps) {
   const i18n = useI18n();
   const {isNavigationCollapsed} = useMediaQuery();
+
+  if (additionalNavigation && process.env.NODE_ENV === 'development') {
```

Comment:
Need to remove the [destructure additionalNavigation prop](https://github.com/Shopify/polaris-react/pull/5167/files#diff-72cf0568af887cf1400f47f978cf95fd4eb29433c47df27968dc6e25355ef533R76) above